plugins/modules/python_requirements_info.py

```diff
@@ -174,13 +175,16 @@ def main():
     )
 
     for dep in module.params['dependencies']:
+        dep = dep.replace(' ', '')
```

Comment:
Ah, that explains why you are disregarding whitespaces in the regexp, but I would expect that to have consequences.

If suppose this test would fail:
```yaml
- name: Ensure sampleprojectpy2 is installed
  python_requirements_info:
    dependencies: sampleprojectpy2     [ test ] >=       1.3.1
  register: _py_facts
  failed_when: |
      ('sampleprojectpy2     [ test ]' not in _py_facts.valid)
```
because all the keys in the dict `<output>.valid` have been "normalised to no-spaces", right? 
I cannot think of a better way to handle that in the code right now, but this should definitely be mentioned in the RETURN documentation block.

Reply:
That is correct, I'll amend the doc